sandbox/anvi-script-filter-hmm-hit-table

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+import sys
+import os
+import numpy as np
+import pandas as pd
+
+import anvio
+import anvio.data.hmm
+import anvio.db as db
+import anvio.hmmops as hmmops
+import anvio.bamops as bamops
+import anvio.terminal as terminal
+import anvio.filesnpaths as filesnpaths
+
+from anvio.dbops import ContigsDatabase
+from anvio.parsers import parser_modules
+from anvio.argparse import ArgumentParser
+from anvio.tables.hmmhits import TablesForHMMHits
+from anvio.errors import ConfigError, FilesNPathsError
+
+__author__ = "Developers of anvi'o (see AUTHORS.txt)"
+__copyright__ = "Copyleft 2015-2020, the Meren Lab (http://merenlab.org/)"
+__credits__ = []
+__license__ = "GPL 3.0"
+__version__ = anvio.__version__
+__maintainer__ = "Matthew S. Schechter"
+__email__ = "mschechter@uchicago.edu"
+__provides__ = ["contigs-db", "hmm-hits"]
+__requires__ = ["contigs-db","hmm-source", "hmm-hits"]
+__description__ = ("Add or remove entries in a contigDB hmm_hits table.")
+
+pp = terminal.pretty_print
+
+class FilterHmmHitsTable(object):
+    """A class to filter hmm_hits from domtblout from hmmsearch."""
+    def __init__(self, args, quiet=False, run=terminal.Run(), progress=terminal.Progress()):
+        self.args = args
+        self.run = run
+        self.progress = progress
+
+        A = lambda x: args.__dict__[x] if x in args.__dict__ else None
+        self.contigs_db_path=A("contigs_db")
+        self.domtblout=A("domtblout")
+        self.hmm_source=A("hmm_source")
+        self.target_coverage=A("target_coverage")
+        self.query_coverage=A("query_coverage")
+        self.list_hmm_sources=A("list_hmm_sources")
+
+        if self.list_hmm_sources:
+            ContigsDatabase(self.contigs_db_path).list_available_hmm_sources()
+            sys.exit()
+
+        self.sanity_checks()
+
+        self.import_domtblout()
+
+        x = self.filter_domtblout()
+
+        self.get_hmm_hit_table(hmmsearch_tbl=x)
+
+        # remove the tmp file
+        os.remove(x)
+
+
+    def sanity_checks(self):
+
+        filesnpaths.is_file_exists(self.contigs_db_path)
+        self.run.info("Database Path", self.contigs_db_path)
+
+        if not self.hmm_source:
+            raise ConfigError("Please provide a hmm-source :)")
+
+        info_table = hmmops.SequencesForHMMHits(self.contigs_db_path).hmm_hits_info
+
+        if self.hmm_source not in info_table:
+            raise ConfigError("Whoa there, the HMM source "'%s'" is not in your contigsDB: %s "
+                              "Please try re-running anvi-run-hmms and make sure to specify you HMM source of interest." % (self.hmm_source, self.contigs_db_path))
+
```

Comment:
You probably don't know about this convention yet, but a few months(?) ago we decided to start using python formatted strings (f-strings) instead of this %s replacement syntax. It is a bit more readable :) I will update some of these in a sec so you can see how it looks like, but no worries if you don't feel like changing the rest (we left all our old %s code alone when we decided to switch) - it's just one more thing to think about for next time!

Reply:
Thanks for pointing that out @ivagljiva, I'll change all of them to f-strings to practice the syntax :)